com.hazelcast.jet.hadoop.file.JsonFileFormatTest.shouldThrowWhenWrongFormatting

_master_ (commit a1938f1e63b36c8977848aad8b08fd1a0d316e22)

Failed on IBM JDK 8: http://jenkins.hazelcast.com/job/jet-oss-master-ibm-jdk8/336/testReport/junit/com.hazelcast.jet.hadoop.file/JsonFileFormatTest/shouldThrowWhenWrongFormatting_1__useHadoop_false_/

Stacktrace:
```
java.lang.AssertionError: 

Expecting a throwable with root cause being an instance of:
 <com.fasterxml.jackson.core.JsonParseException>
but was an instance of:
 <com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:304)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:822)
Caused by: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.jr.ob.ValueIterator._handleIOException(ValueIterator.java)
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java)
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823)
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235)
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235)
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235)
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51)
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247)
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405)
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421)
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293)
	... 5 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:707)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:632)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipComma(ReaderBasedJsonParser.java:2324)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:921)
	at com.fasterxml.jackson.jr.ob.impl.BeanReader.read(BeanReader.java)
	at com.fasterxml.jackson.jr.ob.impl.JSONReader.readBean(JSONReader.java)
	at com.fasterxml.jackson.jr.ob.ValueIterator.nextValue(ValueIterator.java)
	... 19 more
>
	at com.hazelcast.jet.hadoop.file.BaseFileFormatTest.assertJobFailed(BaseFileFormatTest.java:115)
	at com.hazelcast.jet.hadoop.file.JsonFileFormatTest.shouldThrowWhenWrongFormatting(JsonFileFormatTest.java:128)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:62)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:822)
```

Standard output:
```
2021-03-02 04:36:35,096 [ INFO] [Time-limited test] [c.h.i.m.i.MetricsConfigHelper]: [LOCAL] [jet] [4.5-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
2021-03-02 04:36:35,141 [ INFO] [Time-limited test] [c.h.system]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Hazelcast Jet 4.5-SNAPSHOT (20210302 - a1938f1) starting at [127.0.0.1]:5701
2021-03-02 04:36:35,141 [ INFO] [Time-limited test] [c.h.system]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Based on Hazelcast IMDG version: 4.2.0 (20210301 - 09e9877)
2021-03-02 04:36:35,141 [ INFO] [Time-limited test] [c.h.system]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Cluster name: jet
2021-03-02 04:36:35,141 [ INFO] [Time-limited test] [c.h.system]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] 
	o   o   o   o---o o---o o     o---o   o   o---o o-o-o        o o---o o-o-o
	|   |  / \     /  |     |     |      / \  |       |          | |       |
	o---o o---o   o   o-o   |     o     o---o o---o   |          | o-o     |
	|   | |   |  /    |     |     |     |   |     |   |      \   | |       |
	o   o o   o o---o o---o o---o o---o o   o o---o   o       o--o o---o   o
2021-03-02 04:36:35,141 [ INFO] [Time-limited test] [c.h.system]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
2021-03-02 04:36:35,148 [ INFO] [Time-limited test] [c.h.i.m.i.MetricsConfigHelper]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
2021-03-02 04:36:35,159 [ WARN] [Time-limited test] [c.h.c.CPSubsystem]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
2021-03-02 04:36:35,178 [ INFO] [Time-limited test] [c.h.j.i.JetService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Setting number of cooperative threads and default parallelism to 72
2021-03-02 04:36:35,188 [ INFO] [Time-limited test] [c.h.i.d.Diagnostics]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
2021-03-02 04:36:35,189 [ INFO] [Time-limited test] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is STARTING
2021-03-02 04:36:35,191 [ INFO] [Time-limited test] [c.h.i.c.ClusterService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 0719067c-b333-4c8a-835a-1ce165aa9d02 this
]

2021-03-02 04:36:35,192 [ INFO] [Time-limited test] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is STARTED
2021-03-02 04:36:35,195 [DEBUG] [hz.silly_keldysh.cached.thread-3] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
2021-03-02 04:36:35,195 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.i.p.i.PartitionStateManager]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Initializing cluster partition table arrangement...
2021-03-02 04:36:35,197 [ INFO] [hz.silly_keldysh.HealthMonitor] [c.h.i.d.HealthMonitor]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] processors=72, physical.memory.total=377.6G, physical.memory.free=97.7G, swap.space.total=4.0G, swap.space.free=4.0G, heap.memory.used=179.7M, heap.memory.free=844.3M, heap.memory.total=1024.0M, heap.memory.max=1024.0M, heap.memory.used/total=17.54%, heap.memory.used/max=17.54%, minor.gc.count=6, minor.gc.time=337ms, major.gc.count=0, major.gc.time=0ms, load.process=3.99%, load.system=73.19%, load.systemAverage=73.55, thread.count=292, thread.peakCount=298, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=1, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=6, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
2021-03-02 04:36:35,211 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.p.Planner]: Watermarks in the pipeline will be throttled to 1000
2021-03-02 04:36:35,216 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Starting job 05d0-8d2a-3e00-0001 based on submit request
2021-03-02 04:36:35,222 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Didn't find any snapshot to restore for job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,222 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Start executing job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001, execution graph in DOT format:
digraph DAG {
	"files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false" [localParallelism=4];
	"loggerSink" [localParallelism=1];
	"files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false" -> "loggerSink" [queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
2021-03-02 04:36:35,223 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Building execution plan for job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,224 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Built execution plans for job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,225 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.o.InitExecutionOperation]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Initializing execution plan for job 05d0-8d2a-3e00-0001, execution 05d0-8d2a-3e01-0001 from [127.0.0.1]:5701
2021-03-02 04:36:35,231 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Execution plan for jobId=05d0-8d2a-3e00-0001, jobName='05d0-8d2a-3e00-0001', executionId=05d0-8d2a-3e01-0001 initialized
2021-03-02 04:36:35,232 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Init of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 was successful
2021-03-02 04:36:35,232 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Executing job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,232 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Start execution of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 from coordinator [127.0.0.1]:5701
2021-03-02 04:36:35,238 [ WARN] [hz.silly_keldysh.jet.blocking.thread-2] [c.h.j.i.e.TaskletExecutionService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}
java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.jr.ob.ValueIterator._handleIOException(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823) ~[?:1.8.0]
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) [?:1.8.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:1.8.0]
	at java.lang.Thread.run(Thread.java:822) [?:2.9 (07-24-2020)]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:707) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:632) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipComma(ReaderBasedJsonParser.java:2324) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:921) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.BeanReader.read(BeanReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.JSONReader.readBean(JSONReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.nextValue(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	... 19 more
2021-03-02 04:36:35,241 [DEBUG] [hz.silly_keldysh.jet.cooperative.thread-1] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Execution of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:303) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:618) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:602) ~[?:1.8.0]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:499) [?:1.8.0]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2001) [?:1.8.0]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:59) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:461) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.dismissTasklet(TaskletExecutionService.java:392) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:378) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker$$Lambda$1217/0000000008D008B0.accept(Unknown Source) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:906) [?:1.8.0]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:346) [hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:822) [?:2.9 (07-24-2020)]
Caused by: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:304) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]
	... 1 more
Caused by: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.jr.ob.ValueIterator._handleIOException(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823) ~[?:1.8.0]
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]
	... 1 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:707) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:632) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipComma(ReaderBasedJsonParser.java:2324) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:921) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.BeanReader.read(BeanReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.JSONReader.readBean(JSONReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.nextValue(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823) ~[?:1.8.0]
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]
	... 1 more
2021-03-02 04:36:35,245 [DEBUG] [ForkJoinPool.commonPool-worker-57] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Execution of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 has failures: [[127.0.0.1]:5701=com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]]
2021-03-02 04:36:35,246 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Sending CompleteExecutionOperation for job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,246 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.o.CompleteExecutionOperation]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Completing execution 05d0-8d2a-3e01-0001 from caller [127.0.0.1]:5701, error=com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
2021-03-02 04:36:35,246 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Completed execution of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001
2021-03-02 04:36:35,247 [ERROR] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Execution of job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 failed
	Start time: 2021-03-02T04:36:35.216
	Duration: 00:00:00.031
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
com.hazelcast.jet.JetException: Exception in ProcessorTasklet{05d0-8d2a-3e00-0001/files(path=/home/jenkins/jenkins_slave/workspace/jet-oss-master-ibm-jdk8/extensions/hadoop/src/test/resources, glob=file-invalid.jsonl, hadoop=false#2}: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:304) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]
	at java.lang.Thread.run(Thread.java:822) ~[?:2.9 (07-24-2020)]
Caused by: java.lang.RuntimeException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.jr.ob.ValueIterator._handleIOException(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823) ~[?:1.8.0]
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	... 5 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('{' (code 123)): was expecting comma to separate Object entries
 at [Source: (InputStreamReader); line: 2, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:707) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:632) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipComma(ReaderBasedJsonParser.java:2324) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:921) ~[jackson-core-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.BeanReader.read(BeanReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.impl.JSONReader.readBean(JSONReader.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.nextValue(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at com.fasterxml.jackson.jr.ob.ValueIterator.next(ValueIterator.java) ~[jackson-jr-objects-2.11.2.jar:2.11.2]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823) ~[?:1.8.0]
	at com.hazelcast.jet.Traversers$SpliteratorTraverser.next(Traversers.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.Traverser$5.next(Traverser.java:235) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:51) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP$LocalFileTraverser.next(ReadFilesP.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.ReadFilesP.complete(ReadFilesP.java:116) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:472) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:372) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:247) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:293) ~[hazelcast-jet-core-4.5-SNAPSHOT.jar:4.5-SNAPSHOT]
	... 5 more
2021-03-02 04:36:35,251 [DEBUG] [hz.silly_keldysh.cached.thread-2] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] job '05d0-8d2a-3e00-0001', execution 05d0-8d2a-3e01-0001 is completed
2021-03-02 04:36:35,259 [DEBUG] [Time-limited test] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Added a shutting-down member: 0719067c-b333-4c8a-835a-1ce165aa9d02
2021-03-02 04:36:35,260 [ INFO] [Time-limited test] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
2021-03-02 04:36:35,261 [ INFO] [hz.silly_keldysh.cached.thread-2] [c.h.i.p.i.MigrationManager]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - 0719067c-b333-4c8a-835a-1ce165aa9d02 this is handled
2021-03-02 04:36:35,262 [ INFO] [Time-limited test] [c.h.i.i.Node]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Shutting down connection manager...
2021-03-02 04:36:35,262 [ INFO] [Time-limited test] [c.h.i.i.Node]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Shutting down node engine...
2021-03-02 04:36:35,288 [ INFO] [Time-limited test] [c.h.i.i.NodeExtension]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Destroying node NodeExtension.
2021-03-02 04:36:35,289 [ INFO] [Time-limited test] [c.h.i.i.Node]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Hazelcast Shutdown is completed in 27 ms.
2021-03-02 04:36:35,289 [ INFO] [Time-limited test] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
2021-03-02 04:36:35,290 [ INFO] [Time-limited test] [c.h.j.c.JetTestSupport]: Terminating instanceFactory in JetTestSupport.@After
2021-03-02 04:36:35,291 [ INFO] [Thread-36] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
2021-03-02 04:36:35,291 [ INFO] [Thread-36] [c.h.i.i.Node]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] Node is already shutting down... Waiting for shutdown process to complete...
2021-03-02 04:36:35,291 [ INFO] [Thread-36] [c.h.c.LifecycleService]: [127.0.0.1]:5701 [jet] [4.5-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after shouldThrowWhenWrongFormatting[1: useHadoop=false](com.hazelcast.jet.hadoop.file.JsonFileFormatTest): BuildInfo{version='4.2-SNAPSHOT', build='20210301', buildNumber=20210301, revision=09e9877, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='4.5-SNAPSHOT', build='20210302', revision='a1938f1'}}
Hiccups measured while running test 'shouldThrowWhenWrongFormatting[1: useHadoop=false](com.hazelcast.jet.hadoop.file.JsonFileFormatTest):'
04:36:35, accumulated pauses: 2 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

--- a/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/AvroFileFormatTest.java
+++ b/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/AvroFileFormatTest.java
@@ -23,6 +23,7 @@ import com.hazelcast.jet.pipeline.file.FileFormat;
 import com.hazelcast.jet.pipeline.file.FileSourceBuilder;
 import com.hazelcast.jet.pipeline.file.FileSources;
 import org.apache.avro.AvroTypeException;
+import org.apache.avro.InvalidAvroMagicException;
 import org.apache.avro.file.DataFileWriter;
 import org.apache.avro.specific.SpecificDatumWriter;
 import org.apache.commons.io.FileUtils;
@@ -79,7 +80,7 @@ public class AvroFileFormatTest extends BaseFileFormatTest {
                                                     .glob("invalid-data.png")
                                                     .format(FileFormat.avro(User.class));
 
-        assertJobFailed(source, IOException.class, "Not an Avro data file");
+        assertJobFailed(source, InvalidAvroMagicException.class, "Not an Avro data file");
     }
 
     @Test

--- a/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/BaseFileFormatTest.java
+++ b/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/BaseFileFormatTest.java
@@ -112,7 +112,8 @@ public abstract class BaseFileFormatTest extends HadoopTestSupport {
         try {
             assertThatThrownBy(() -> jets[0].newJob(p).join())
                     .hasCauseInstanceOf(JetException.class)
-                    .hasRootCauseInstanceOf(expectedRootException)
+                    // can't use hasRootCauseInstanceOf because of mismatch between shaded/non-shaded version
+                    .hasStackTraceContaining(expectedRootException.getName())
                     .hasMessageContaining(expectedMessage);
         } finally {
             for (JetInstance jet : jets) {

--- a/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/JsonFileFormatTest.java
+++ b/extensions/hadoop/src/test/java/com/hazelcast/jet/hadoop/file/JsonFileFormatTest.java
@@ -17,6 +17,7 @@
 package com.hazelcast.jet.hadoop.file;
 
 import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.io.JsonEOFException;
 import com.google.common.collect.ImmutableMap;
 import com.hazelcast.jet.hadoop.file.model.User;
 import com.hazelcast.jet.pipeline.file.FileFormat;
@@ -150,6 +151,6 @@ public class JsonFileFormatTest extends BaseFileFormatTest {
                                                     .glob("file-multiline.jsonl")
                                                     .format(FileFormat.json(User.class).multiline(false));
 
-        assertJobFailed(source, JsonParseException.class, "");
+        assertJobFailed(source, JsonEOFException.class, "expected close marker for Object");
     }
 }

